tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/MuleClassPathClassifier.java

```diff
@@ -0,0 +1,394 @@
+        generatedResourcesDirectory.mkdirs();
+        ExtensionsTestInfrastructureDiscoverer extensionDiscoverer = new ExtensionsTestInfrastructureDiscoverer(createExtensionManager(), generatedResourcesDirectory);
+        extensionDiscoverer.discoverExtensions(new Describer[0], new Class[] {extension});
+        try
+        {
+            // Registering parent file as resource to be used from the configuration builder
+            extensionURLs.add(generatedResourcesDirectory.getParentFile().toURI().toURL());
+        }
+        catch (MalformedURLException e)
+        {
+            throw new IllegalArgumentException("Error while building resource URL for directory: " + generatedResourcesDirectory.getPath(), e);
+        }
+
+        int sizeBeforeDepResolver = extensionURLs.size();
+
+        new DependencyResolver(new ConfigurationBuilder()
+                                       .setMavenDependencyGraph(allDependencies)
+                                       .includeRootArtifact(rootArtifact -> rootArtifact.getArtifactId().equals(extensionMavenArtifactId.toString()))
+                                       .selectDependencies(
+                                               new DependenciesFilterBuilder()
+                                                       .match(dependency -> dependency.getArtifactId().equals(extensionMavenArtifactId.toString())
+                                                                            || (compileArtifact.getArtifactId().equals(extensionMavenArtifactId.toString()) && dependency.isCompileScope() && !exclusion.test(dependency)))
+                                       )
+                                       .collectTransitiveDependencies(
+                                               new TransitiveDependenciesFilterBuilder()
+                                                       .match(transitiveDependency -> transitiveDependency.isCompileScope() && !exclusion.test(transitiveDependency))
+                                       )).resolveDependencies().stream().filter(d -> !d.isPomType()).map(dependency -> artifactToClassPathURLResolver.resolveURL(dependency, classPathURLs)).forEach(extensionURLs::add);
+
+        if (extensionURLs.size() == sizeBeforeDepResolver)
+        {
+            throw new IllegalStateException("There should be at least one compile dependency found that matched to extension: '" + extension.getName() +
+                                            "'. Be aware that compile scope is what the classification uses for selecting the urls to be added to plugin class loader");
+        }
+        return new ExtensionPluginUrlClassification(extension, extensionURLs);
+    }
+
+    /**
+     * Does the classification where the rootArtifact is not an extension so if it is not an extension it is treated as a plugin
```

Comment:
Does the classification where the rootArtifact is not an extension so if it is not an extension it is treated as a plugin > Classifies URLs for a plugin that is not an extension